scripts/bin/graphics-core22-provider-wrapper

```diff
@@ -0,0 +1,50 @@
+#!/bin/bash
+set -euo pipefail
+
+SELF="$( cd -- "$(dirname "$0")/.." ; pwd -P )"
+
+if [ "$SNAP_ARCH" == "amd64" ]; then
+  ARCH_TRIPLETS=( "x86_64-linux-gnu" "i386-linux-gnu" )
+  GCONV_PATH=${GCONV_PATH:+$GCONV_PATH:}${SELF}/lib/i386-linux-gnu/gconv
+elif [ "$SNAP_ARCH" == "armhf" ]; then
+  ARCH_TRIPLETS=( "arm-linux-gnueabihf" )
+else
+  ARCH_TRIPLETS=( "$(arch)-linux-gnu" )
+fi
```

Comment:
the paths in `/usr/lib/<thing>` is not based on `arch` call, but rather is `dpkg-architecture -qDEB_HOST_MULTIARCH -a$SNAP_ARCH`. Which during snapcraft time is known as SNAPCRAFT_ARCH_TRIPLET.

Obviously vendoring dpkg-architecture is pain, and snapcraft is not available at runtime.

I would rather want to see https://github.com/MirServer/graphics-core22/blob/main/bin/graphics-core22-wrapper have a `case $SNAP_ARCH` and set appropriate SNAPCRAFT_ARCH_TRIPLET. Such that provider wrappers can use $SNAPCRAFT_ARCH_TRIPLET throughout their provider script.

Reply:
So you're saying that someone could, arguably, install a arm64 snap on amd64, and that case should be handled?

I want to keep graphics-core22-wrapper minimal, as that one's vendored into consumer snaps, so any changes there would require rebuilds of all the snaps. I can likely handle the case above in the `-provider-wrapper`, though.

Reply:
no not arm64 and amd64.

core* base snap on amd64 contains two ld.so and supports to run x86_64 and i686 binaries only.
core* base snap on arm64 contains two ld.so and supports to run aarch64 and armhf binaries only.
all other core snaps on all other snap arches can only run a native arch.

So :i386 binaries should be vendored on arm64 snap arch only.

And one may want to do :armhf binaries on arm64 snap arch too. However, for now, one probably will have everything armhf. Until we like kill armhf (well, not really possible)

Reply:
this is probably more a use case for a mesa-core22 "fat" track which would include foreign arch binaries.

Reply:
> So :i386 binaries should be vendored on arm64 snap arch only.

On `amd64` you mean? ;)

> this is probably more a use case for a mesa-core22 "fat" track which would include foreign arch binaries.

Correct, this may well not be the default.

So there's two options I can see:
1. ```bash
   case $SNAP_ARCH in
   amd64)
     ARCH_TRIPLETS=( "x86_64-linux-gnu" "i386-linux-gnu" );;
   arm64)
     ARCH_TRIPLETS=( "aarch64-linux-gnu" "arm-linux-gnueabihf" );;
   armhf)
   # ...
   ```

2. ```bash
   ARCH_TRIPLETS=( %ARCH_TRIPLETS% )
   ```
   And replace that with the right thing based on `$CRAFT_ARCH_TRIPLET` build-time.

Reply:
lol, yes typpo.

I like #2. option a lot. Because indeed only the matching one matters at runtime. and one doesn't need to know about any other arches.

Reply:
> I like #2. option a lot. Because indeed only the matching one matters at runtime. and one doesn't need to know about any other arches.

Done. Note in multi-arch snaps, like this one, it still does require both arches in there.

Reply:
Resolved.